packages/kamino-sdk/tests/stratCreation.test.ts

```diff
@@ -506,4 +506,111 @@ describe('Kamino strategy creation SDK Tests', () => {
+    );
+
+    let ixs: TransactionInstruction[] = [];
+    ixs.push(createRaydiumStrategyAccountIx);
+    ixs.push(buildNewStrategyIxs[0]);
+    const createStratTx = await kamino.getTransactionV2Message(signer.publicKey, ixs);
+    const createStratTransactionV0 = new VersionedTransaction(createStratTx);
+    createStratTransactionV0.sign([newStrategy, signer]);
+    //@ts-ignore
+    let txHash = await sendAndConfirmTransaction(kamino._connection, createStratTransactionV0);
+    console.log('create strategy tx hash', txHash);
+
+    let strategySetupIxs: TransactionInstruction[] = [];
+    buildNewStrategyIxs[1].slice(0, 4).map((ix) => strategySetupIxs.push(ix));
+    const setupStratTx = await kamino.getTransactionV2Message(signer.publicKey, strategySetupIxs);
+    const setupStratTransactionV0 = new VersionedTransaction(setupStratTx);
+    setupStratTransactionV0.sign([signer]);
+
+    //@ts-ignore
+    txHash = await sendAndConfirmTransaction(kamino._connection, setupStratTransactionV0);
+    console.log('setup strategy tx hash', txHash);
+
+    let strategySetupFeesIxs: TransactionInstruction[] = [];
+    buildNewStrategyIxs[1].slice(4).map((ix) => strategySetupFeesIxs.push(ix));
+    strategySetupFeesIxs.push(buildNewStrategyIxs[2]);
+    const setupStratFeesTx = await kamino.getTransactionV2Message(signer.publicKey, strategySetupFeesIxs);
+    const setupStratFeesTransactionV0 = new VersionedTransaction(setupStratFeesTx);
+    setupStratFeesTransactionV0.sign([signer]);
+    //@ts-ignore
+    txHash = await sendAndConfirmTransaction(kamino._connection, setupStratFeesTransactionV0);
+    console.log('setup strategy fees tx hash', txHash);
+
+    let lowerPriceBpsDifference = new Decimal(10.0);
+    let upperPriceBpsDifference = new Decimal(11.0);
+
+    // update rebalance params
+    let updateRebalanceParamsIx = await kamino.getUpdateRebalancingParmsIxns(
+      signer.publicKey,
+      newStrategy.publicKey,
+      [lowerPriceBpsDifference, upperPriceBpsDifference],
+      new RebalanceType.PricePercentage()
+    );
+    let tx = createTransactionWithExtraBudget(signer.publicKey);
+    tx.add(updateRebalanceParamsIx);
+    let updateRebalanceParamsTxHash = await sendTransactionWithLogs(connection, tx, signer.publicKey, [signer]);
+    console.log('update Rebalance Params Tx Hash ', updateRebalanceParamsTxHash);
+
+    let strategyData = await kamino.getStrategies([newStrategy.publicKey]);
+    expect(strategyData[0]?.rebalanceRaw[0] == lowerPriceBpsDifference);
+    expect(strategyData[0]?.rebalanceRaw[2] == upperPriceBpsDifference);
+
+    // open position
+    let pool = strategyData[0]?.pool;
+    if (!pool) {
+      throw new Error(`Whirlpool ${pool} cannot be found in strategy`);
+    }
+
+    let poolData = await kamino.getWhirlpoolByAddress(pool);
+    if (!poolData) {
+      throw new Error(`Unable to fetch Whirlpool ${pool}`);
+    }
+
+    const newPosition = Keypair.generate();
```

Comment:
Please add this as part of the create strategy SDK, we don't need this just in the tests, we want this as part of the SDK, so the sdk needs to open the position already.

do the same for percentage based as well as for manual based.

Reply:
I have created a new sdk method for this (`openPositionFromRebalanceParameters`), as opening position requires `baseVaultAutority` that is created when the strategy is created, so i cannot access it before strat is created